packages/language-server/src/plugins/typescript/features/CodeActionsProvider.ts

```diff
@@ -77,12 +80,23 @@ export class CodeActionsProviderImpl implements CodeActionsProvider {
             return [];
         }
 
+        const useSemicolons =
+            this.configManager.getMergedPrettierConfig(
+                await importPrettier(document.getFilePath()!).resolveConfig(
+                    document.getFilePath()!,
+                    {
+                        editorconfig: true
+                    }
+                )
+            ).semi ?? true;
         const changes = lang.organizeImports(
             {
                 fileName: tsDoc.filePath,
                 type: 'file'
             },
-            {},
+            {
+                semicolons: useSemicolons
```

Comment:
Seems like it's because this config is not boolean but 'ignore', 'insert' and 'remove'. Where 'ignore' and falsy is auto-detect.

Reply:
Thank you, that was it! Should've looked at the option more closely 😄